frontend/vite.config.ts

```diff
@@ -6,10 +6,32 @@ import react from "@vitejs/plugin-react"
 import svgr from "vite-plugin-svgr"
 import fs from "fs"
 import { viteStaticCopy } from "vite-plugin-static-copy"
-
-// import viteReact from '@vitejs/plugin-react'
 import { TanStackRouterVite } from "@tanstack/router-plugin/vite"
 
+// 🔹 Add this helper plugin at the top
+function injectDevConfig() {
+  return {
+    name: "inject-dev-config",
+    transformIndexHtml(html: string) {
+      if (process.env.NODE_ENV === "development") {
+        return html.replace(
+          "</head>",
+          `
+          <script>
+            window.CONFIG = {
+              API_BASE_URL: "${process.env.VITE_API_BASE_URL || "http://localhost:3000"}",
```

Comment:
```suggestion
              API_BASE_URL: "${process.env.VITE_API_BASE_URL}",
```
we should not put default "http://localhost:3000" in this file during generation, so that we can use window.location.origin for fallback if `API_BASE_URL` is empty